### ISSUE TITLE
Add a damageable item function

### ISSUE THREAD

**Issue by Khristian-kun** [zhuowei/MCPELauncher#523]
The most of the modpe modders makes items looking as tools, and when you use the item with enchantments it is reset

**Comment by Black-DeathPE** [comment]
Item.setMaxDamage()?

**Comment by Khristian-kun** [comment]
@Black-DeathPE no, that only defines the maxdamage, what i was saying is that when you use the item, it dont takes damage, and if you do it with a custom function, the enchantment dissapears

**Comment by zhuowei** [comment]
So I guess there's two things I need to do:
- add some sort of way to create a tool item that damages when you hit it
and
- add some sort of way to keep the enchantment when calling Entity.setCarriedItem or setInventorySlot

**Comment by Khristian-kun** [comment]
exactly

**Comment by InusualZ** [comment]
@zhuowei Actually, I think there is a base class for all the tool you could use that I think
